tests/components/remote_calendar/test_calendar.py

```diff
@@ -0,0 +1,479 @@
+            END:VEVENT
+            END:VCALENDAR
+            """
+        ),
+    ],
+    indirect=True,
+)
+async def test_recurring_event(
+    mock_httpx_client: AsyncMock,
+    get_events: GetEventsFn,
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+) -> None:
+    """Test an event with a recurrence rule."""
+    await setup_integration(hass, config_entry)
+
+    events = await get_events("2022-08-20T00:00:00", "2022-09-20T00:00:00")
+    assert list(map(event_fields, events)) == [
+        {
+            "summary": "Monday meeting",
+            "start": {"dateTime": "2022-08-29T09:00:00-06:00"},
+            "end": {"dateTime": "2022-08-29T10:00:00-06:00"},
+            "recurrence_id": "20220829T090000",
+        },
+        {
+            "summary": "Monday meeting",
+            "start": {"dateTime": "2022-09-05T09:00:00-06:00"},
+            "end": {"dateTime": "2022-09-05T10:00:00-06:00"},
+            "recurrence_id": "20220905T090000",
+        },
+        {
+            "summary": "Monday meeting",
+            "start": {"dateTime": "2022-09-12T09:00:00-06:00"},
+            "end": {"dateTime": "2022-09-12T10:00:00-06:00"},
+            "recurrence_id": "20220912T090000",
+        },
+        {
+            "summary": "Monday meeting",
+            "start": {"dateTime": "2022-09-19T09:00:00-06:00"},
+            "end": {"dateTime": "2022-09-19T10:00:00-06:00"},
+            "recurrence_id": "20220919T090000",
+        },
+    ]
+
+
+@pytest.mark.parametrize(
+    "ics_content",
+    [
+        textwrap.dedent(
+            """\
+            BEGIN:VCALENDAR
+
+            BEGIN:VEVENT
+            DTSTART:20221008
+            DTEND:20221009
+            SUMMARY:All Day Event
+            END:VEVENT
+
+            BEGIN:VEVENT
+            DTSTART:20221007T230000Z
+            DTEND:20221008T233000Z
+            SUMMARY:One
+            END:VEVENT
+
+            BEGIN:VEVENT
+            DTSTART:20221008T010000Z
+            DTEND:20221008T020000Z
+            SUMMARY:Two
+            END:VEVENT
+
+            END:VCALENDAR
+            """
+        ),
+    ],
+    indirect=True,
+)
+@pytest.mark.parametrize(
+    ("time_zone", "event_order"),
+    [
+        ("America/Los_Angeles", ["One", "Two", "All Day Event"]),
+        ("America/Regina", ["One", "Two", "All Day Event"]),
+        ("UTC", ["One", "All Day Event", "Two"]),
+        ("Asia/Tokyo", ["All Day Event", "One", "Two"]),
+    ],
+)
+async def test_all_day_iter_order(
+    mock_httpx_client: AsyncMock,
+    get_events: GetEventsFn,
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    event_order: list[str],
+) -> None:
+    """Test the sort order of an all day events depending on the time zone."""
+    await setup_integration(hass, config_entry)
+
+    events = await get_events("2022-10-06T00:00:00Z", "2022-10-09T00:00:00Z")
+    assert [event["summary"] for event in events] == event_order
+
+
+# @pytest.mark.parametrize(
```

Comment:
I think it is OK to not test every corner case here. At some level, the local calendar test is exercising some `ical` functionality to make sure it actually works well, but we can trust that it mostly handles things.  I think you can remove this commented out code.

Reply:
https://github.com/home-assistant/core/pull/138862/commits/9440926b267988b14536d13c578fe2481ba8a419